workflow/examples/waitForEvent.mdx

```diff
@@ -0,0 +1,138 @@
+---
+title: "Waiting for Events"
+---
+
+## Introduction
+
+This example demonstrates how to handle order processing using Upstash Workflow. The workflow waits for an external system to process an order and resumes once it is 'notified'.
+
+## Use Case
+
+Our workflow will:
+
+1. Receive an order request.
+2. Wait for an external event that indicates the order has been processed.
+3. Process the order upon receiving the event.
+4. Send a confirmation email to the user.
+5. Handle timeout scenarios if the event is not received in time.
+6. Demonstrate how to notify the workflow.
+
+## Code Example
+
+```typescript
+import { serve } from "@upstash/workflow/nextjs";
+
+export const { POST } = serve(async (context) => {
+  const { orderId, userEmail } = context.requestPayload;
+
+  // Step 1: Wait for the order to be processed
+  const { eventData, timeout } = await context.waitForEvent(
+    "wait for order processing",
+    `order-${orderId}`,
+    600 // 10 minutes timeout
+  );
```

Comment:
Just a small idea, directly waiting on an event does not makes sense for me. Maybe we can add a `context.call()` step to describe it better.